src/checkers/inference/solver/frontend/Lattice.java

```diff
@@ -0,0 +1,96 @@
+package checkers.inference.solver.frontend;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * Lattice class contains necessary information about qualifier hierarchy for
+ * constraint constraint solving.
+ * 
+ * @author jianchu
+ *
+ */
+public class Lattice {
+
+    private QualifierHierarchy qualHierarchy;
+    public Map<AnnotationMirror, Collection<AnnotationMirror>> subType = AnnotationUtils .createAnnotationMap();
+    public Map<AnnotationMirror, Collection<AnnotationMirror>> superType = AnnotationUtils.createAnnotationMap();
+    public Map<AnnotationMirror, Collection<AnnotationMirror>> incomparableType = AnnotationUtils.createAnnotationMap();
+    public Map<AnnotationMirror, Integer> typeToInt = AnnotationUtils.createAnnotationMap();
+    public Map<Integer, AnnotationMirror> intToType = new HashMap<Integer, AnnotationMirror>();
+    private Set<? extends AnnotationMirror> allTypes;
+    public AnnotationMirror top;
+    public AnnotationMirror bottom;
+    public int numTypes;
+
+    public void configure() {
```

Comment:
I checked the code and seems the only place called this `configure()` method is in `GeneralSolver`:

```java
protected void configureLattice(QualifierHierarchy qualHierarchy) {
        lattice = new Lattice(qualHierarchy);
        lattice.configure();
    }
```

And it seems solver calls `configure()` immediately after initializing `lattice`. So what is the motivation of initial design to separate `configure()` from constructor? Can we have a javadoc on this method?